### PR TITLE
go version 1.15.3

### PR DESCRIPTION
**Pull request by deitch**
This is the latest version of my many-months-earlier attempt to to move to a modern go version.

This has several benefits which we need:

* faster module vending
* better module debugging (`go mod why` and its tools)
* support for the features of go 1.13+
* support for the various performance improvements of go 1.13+ (and 1.15)
* support for all of the libraries which now only support go 1.13+ (e.g. `github.com/containerd/containerd`)

The last one is particularly important, as we have a major issue with tens of thousands of goroutines in the containerd client. This was fixed in containerd, but we cannot import it without being 1.13+ compliant.

Running `go vet` and `go build` had only one error in `pkg/pillar` which is fixed in this PR.

Someone should look at that one line of code.

### CODE REVIEW

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 1]
```diff
@@ -1559,7 +1559,7 @@ func configToStatus(ctx *domainContext, config types.DomainConfig,
 			ds.Devtype = "hdd"
 		}
 		// map from i=1 to xvdb, 2 to xvdc etc
-		ds.Vdev = "xvd" + string(int('a')+i)
+		ds.Vdev = fmt.Sprintf("xvd%d", int('a')+i)
```
Comment (deitch) [inline pkg/pillar/cmd/domainmgr/domainmgr.go:1562]: @eriknordmark and @rvs (and anyone else who touched this), can you check that this is correct? The output should be `xvd97` then `xvd98`, then `xvd99`, etc.?

Comment (eriknordmark) [reply pkg/pillar/cmd/domainmgr/domainmgr.go:1562]: No, the output should be xvda, xvdb, xvdc. Hence the type conversion from into to string.

Comment (deitch) [reply pkg/pillar/cmd/domainmgr/domainmgr.go:1562]: Ah, now I see it. OK, I will fix this.

Comment (deitch) [reply pkg/pillar/cmd/domainmgr/domainmgr.go]: Fixed. Take a look please.

**Comment by mydatascience** [comment]
It's good to reflect the required go version in README

**Comment by deitch** [comment]
> It's good to reflect the required go version in README

Good point @mydatascience ; updated

**Review by rvs — APPROVED** [review]
LGTM. Lets see if the Eden testing passes now.

**Comment by deitch** [comment]
> Lets see if the Eden testing passes now

Yup. And hopefully it doesn't do anything too strange, like bypassing the makefile or the like.

File: pkg/pillar/Dockerfile.in [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Copyright (c) 2018 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
-ARG GOVER=1.12.4
+ARG GOVER=1.13.5
```
Comment (eriknordmark) [inline pkg/pillar/Dockerfile.in:3]: Why is this not 1.15.3?

**Review by eriknordmark — COMMENTED** [review]
Looks like we are using 1.13.5 to build pillar. Is that correct and the intention?

File: pkg/pillar/Dockerfile.in [hunk 3]
```diff
@@ -17,7 +17,7 @@ RUN patch -p1 < patch01-no-pam.diff && \
     patch -p1 < patch02-rotate-raw-key.diff && \
     make && make install
 
-ARG GOVER=1.12.4
+ARG GOVER=1.13.5
```
Comment (eriknordmark) [inline pkg/pillar/Dockerfile.in:20]: Ditto

**Comment by rvs** [comment]
Closing this since #1565 now superseeds this

**Pull request closed without merging**